src/java.base/share/classes/java/util/random/RandomGenerator.java

```diff
@@ -0,0 +1,1486 @@
+         */
+        void leap();
+
+        /**
+         * Returns the distance by which the
+         * {@link LeapableGenerator#leap() leap}() method will leap forward within
+         * the state cycle of this generator object.
+         *
+         * @return the default leap distance (as a {@code double} value)
+         */
+        double leapDistance();
+
+        /**
+         * Returns an effectively unlimited stream of new pseudorandom number
+         * generators, each of which implements the {@link JumpableGenerator}
+         * interface.
+         *
+         * @return a stream of objects that implement the {@link JumpableGenerator} interface
+         *
+         * @implNote It is permitted to implement this method in a manner equivalent to
+         *           {@link LeapableGenerator#leaps(long) leaps}
+         *           ({@link  Long#MAX_VALUE Long.MAX_VALUE}).
+         *           The default implementation produces a sequential stream that  repeatedly
+         *           calls {@link LeapableGenerator#copy() copy}() and {@link LeapableGenerator#leap() leap}()
+         *           on this generator, and the copies become the generators produced by the stream.
+         */
+        default Stream<JumpableGenerator> leaps() {
+            return Stream.generate(this::copyAndLeap).sequential();
+        }
+
+        /**
+         * Returns a stream producing the given {@code streamSize} number of new
+         * pseudorandom number generators, each of which implements the
+         * {@link JumpableGenerator} interface.
+         *
+         * @param streamSize the number of generators to generate
+         *
+         * @return a stream of objects that implement the {@link JumpableGenerator} interface
+         *
+         * @throws IllegalArgumentException if {@code streamSize} is less than zero
+         *
+         * @implNote The default implementation produces a sequential stream that  repeatedly
+         *           calls {@link LeapableGenerator#copy() copy}() and {@link LeapableGenerator#leap() leap}()
+         *           on this generator, and the copies become the generators produced by the stream.
+         */
+        default Stream<JumpableGenerator> leaps(long streamSize) {
+            return leaps().limit(streamSize);
+        }
+
+        /**
+         * Copy this generator, leap this generator forward, then return the
+         * copy.
+         *
+         * @return a copy of this generator object before the leap occurred
+         */
+        default JumpableGenerator copyAndLeap() {
+            JumpableGenerator result = copy();
+            leap();
+            return result;
+        }
+
+    }
+
+    /**
+     * This interface is designed to provide a common protocol for objects that
+     * generate sequences of pseudorandom numbers (or Boolean values) and
+     * furthermore can easily <i>jump</i> to an arbitrarily specified distant
+     * point in the state cycle.
+     *
```

Comment:
The similarity of the first sentence of each of the Jumpable, Leapable, and arbitrarlyJumpable interface is so similar as to obscure the differences. You have to read 25 words in to be able to find the description that makes them different. The italics help but should include the whole of the phrase that distinguishes them.
Alternatively, move the phrase to the beginning of the sentence or drop the redundant phrasing.
"provide a common protocol of objects that generate pseudorandom sequences of numbers of Boolean values", etc.

Reply:
Jumpable:
     * This interface is designed to provide a common protocol for objects that
     * generate pseudorandom sequences of numbers (or Boolean values) and
     * furthermore can easily <i>jump</i> forward, by a moderate amount (ex.
     * 2<sup>64</sup>) to a distant point in the state cycle.

Leapable:
     * This interface is designed to provide a common protocol for objects that
     * generate sequences of pseudorandom numbers (or Boolean values) and
     * furthermore can easily not only jump but also
     * <i>leap</i> forward, by a large amount (ex. 2<sup>128</sup>), to a
     * very distant point in the state cycle.

ArbitrarilyJumpable:
     * This interface is designed to provide a common protocol for objects that
     * generate sequences of pseudorandom numbers (or Boolean values) and
     * furthermore can easily <i>jump</i> forward, by an arbitrary amount, to a
     * distant point in the state cycle.